### PR TITLE
Update documentation about exporting models

### PR DESCRIPTION
**Pull request by dmontagu**
* I changed 'serialise' to 'serialize' (and similar for -ing, etc.) everywhere. The method names included the 'z' so I think that was the way to go, and ultimately I believe there were far fewer occurrences with an 's' than with a 'z'.
* I reordered some of the sections in the exporting models docs in a way that I felt was more cohesive.
* There was one regression relative to V1 that I noticed in the docs, and which I'm not sure was intentional. I'll point it out in a comment on the code. I'm not sure what we should do about it.

### CODE REVIEW

File: docs/usage/exporting_models.md
Comment (dmontagu) [inline docs/usage/exporting_models.md:274]: I have docs ready for this. @adriangb you didn't do any work on this (or run into issues trying) did you? If not, I'm happy to quickly throw together a PR adding this along with docs after this gets merged.

File: docs/usage/exporting_models.md
Comment (dmontagu) [inline docs/usage/exporting_models.md:472]: The line `    user: User = Field(exclude={'username'})` above is actually a type error for the `exclude` argument. It seems we changed how that argument works. I noticed this before, but wasn't quite sure how we want this to work. @samuelcolvin @adriangb thoughts?

This section may need a bigger rework if we aren't going to fix that.

File: docs/usage/exporting_models.md
Comment (adriangb) [reply docs/usage/exporting_models.md:274]: I may have fixed it in #6007 but I didn’t add a type and export it.

**Review by tpdorsey — APPROVED** [review]
Doc portion of this looks good. Approved pending resolution of comments.

File: docs/blog/pydantic-v2.md
Comment (adriangb) [inline docs/blog/pydantic-v2.md:213]: Just for the record, `serialising` is also correct, it's just a UK thing.

File: docs/usage/exporting_models.md
Comment (adriangb) [inline docs/usage/exporting_models.md:77]: I thought for RootModel in JSON there is no `__root__`/`root` key?

**Review by adriangb — APPROVED** [review]

File: docs/usage/exporting_models.md
Comment (lig) [reply docs/usage/exporting_models.md:77]: There is no `root` key in `json`. It's the same as for `model_dump`. So, `RootModel.model_dump()` doesn't have the `root` field and neither `RootModel.model_dump_json()` does.

IMO, this documentation doesn't contradict this.

Comment (dmontagu) [reply docs/usage/exporting_models.md:77]: How would you guys feel about me replacing lines 74-77 with:

```
The `.model_dump_json()` method serializes a model directly to a JSON-encoded string that is equivalent to the result produced by `.model_dump`.
```

I feel like this addresses the confusion, and is more terse to boot. If I get a 👍 I'll push the change (couldn't do a suggestion since deleted lines are in the span)

File: docs/blog/pydantic-v2.md
Comment (dmontagu) [reply docs/blog/pydantic-v2.md:213]: yeah, I just wanted to standardize it, and since we use `serialize` more frequently than `serialise` throughout the codebase I just find-and-replaced everywhere

File: docs/usage/exporting_models.md
Comment (dmontagu) [reply docs/usage/exporting_models.md]: I made the change since I was working on this to resolve conflicts anyway. Happy to reconsider if appropriate

**Pull request merged**
